package.json

```diff
@@ -19,21 +23,24 @@
     "url": "https://github.com/wikimedia/grunt-stylelint/issues"
   },
   "homepage": "https://github.com/wikimedia/grunt-stylelint",
+  "engines": {
+    "node": ">=6",
+    "npm": ">=3"
```

Comment:
These are not even the changes that stylelint uses.

Reply:
See https://github.com/stylelint/stylelint/blob/master/package.json#L39
And I have declared npm with 3.X because if you use Node 6.X, per default, you should use NPM 3.X

Regards

Reply:
No you shouldn't, you should always use the latest release of npm

Reply:
I can understand the confusion:

````
if you use Node 6.X, per default, you should use NPM 3.X
````

I should say:

````
if you use Node 6.X, per default, you should use at least NPM 3.X
````

(My first sentence means: I you use Node 6, you probably use NPM 3, because when we install node, a related version of npm is installed)

Regards

Reply:
Regardless of the Node.js version, you should *always* use the latest npm release, so for the latest v6.14.1, v8.11.1, v9.11.1 Node.js releases you npm v5.8.0 should be used.

Note: Latest releases at time of writing

Reply:
Here the table which describes which version of npm is installed depends on which version of node you use: https://nodejs.org/en/download/releases/

If you install Node.js 6.2.0, you will have npm 3.9.0

You could update (or downgrade) the npm version, of course
And in all cases, the above configuration does not prevent that at all. It means a based alignment, due to a based alignement on the Node version

Regards

Reply:
Realistically you shouldn't define the npm version in `package.json`, [stylelint doesn't](https://github.com/stylelint/stylelint/blob/master/package.json#L39), locally and on the CI you should use the latest npm.

Reply:
And your plugin does nit use "main" and "lib" (main which is very important and in your case produce bug on grunt loader plugin).

That does not mean this is a good practice to do like you

Regards

Reply:
Hi

Here an interesting link: https://itnext.io/node-engines-helping-developers-everywhere-avoid-phantom-bugs-2eef519604b2

Regards